Asset graph view Minified React error #185 when clicking on a asset

### Apache Airflow version

main (development)

### If "Other Airflow 2 version" selected, which one?

main right now

### What happened?

Clicked on an asset in the assets list that is created with `@asset`, was greeted with this:

![Image](https://github.com/user-attachments/assets/381b283b-3c7d-4744-af8c-2e6223380e45)


```
index-CkMuVLPo.js:48 Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at WS (index-CkMuVLPo.js:48:34133)
    at ien (index-CkMuVLPo.js:46:20052)
    at index-CkMuVLPo.js:46:19912
    at index-CkMuVLPo.js:450:1045
    at Set.forEach (<anonymous>)
    at o (index-CkMuVLPo.js:450:1034)
    at setNodes (index-CkMuVLPo.js:455:49942)
    at index-CkMuVLPo.js:450:6734
    at y0e (index-CkMuVLPo.js:48:24480)
    at r$ (index-CkMuVLPo.js:48:42843)
    at _Jn (index-CkMuVLPo.js:48:41659)
    at YO (index-CkMuVLPo.js:48:40679)
    at W$t (index-CkMuVLPo.js:48:37230)
    at I_ (index-CkMuVLPo.js:46:3323)
    at index-CkMuVLPo.js:48:34565
```

### What you think should happen instead?

An asset graph

### How to reproduce

Use this chain of 3 assets that run normally, run them once:

```
from airflow.sdk import asset


@asset(schedule="@daily")  # A
def extracted_data():  # B
    pass  # C


@asset(schedule=extracted_data)  # D
def transformed_data():
    pass  # E


@asset(schedule=transformed_data)
def loaded_data():
    pass  # F
```

Click on the Assets tab in the sidebar, then on the "loaded_data" asset.

see the error.

### Operating System

MacOs and caffeine

### Versions of Apache Airflow Providers

_No response_

### Deployment

Other

### Deployment details

breeze - now with uv!

### Anything else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

--- a/airflow-core/src/airflow/ui/src/queries/useDependencyGraph.ts
+++ b/airflow-core/src/airflow/ui/src/queries/useDependencyGraph.ts
@@ -43,7 +43,7 @@ export const useDependencyGraph = (
     const key = UseDependenciesServiceGetDependenciesKeyFn({ nodeId: node.id });
     const queryData = queryClient.getQueryData(key);
 
-    if (!Boolean(queryData) && Boolean(query.data)) {
+    if (!Boolean(queryData)) {
       queryClient.setQueryData(key, query.data);
     }
   });

--- a/airflow-core/src/airflow/ui/src/queries/useDependencyGraph.ts
+++ b/airflow-core/src/airflow/ui/src/queries/useDependencyGraph.ts
@@ -40,7 +40,12 @@ export const useDependencyGraph = (
 
   // Update the queries for all connected assets and dags so we save an API request
   query.data?.nodes.forEach((node) => {
-    queryClient.setQueryData(UseDependenciesServiceGetDependenciesKeyFn({ nodeId: node.id }), query.data);
+    const key = UseDependenciesServiceGetDependenciesKeyFn({ nodeId: node.id });
+    const queryData = queryClient.getQueryData(key);
+
+    if (!Boolean(queryData) && Boolean(query.data)) {
+      queryClient.setQueryData(key, query.data);
+    }
   });
 
   return query;

